aten/src/ATen/SparseCsrTensorImpl.h

```diff
@@ -57,6 +57,22 @@ struct TORCH_API SparseCsrTensorImpl : public TensorImpl {
     return col_indices_.size(-1);
   }
 
+  int64_t batch_dim() const {
+    return crow_indices_.dim() - 1;
+  }
+
+  int64_t block_dim() const {
```

Comment:
nit: I'd make this a private helper function, since blocks don't contribute to the dimension of a Tensor.

Reply:
The suggestion is applied.

Re unifying CSR/CSC with BSR/BSC: implementation-wise, these are unified. What could be done is to consider CSR/CSC as a special case of BSR/BSC with `blocksize=(1, 1)` and drop `CSR/CSC` layouts. That would make sense if the underlying libraries would provide full support to block formats but currently, the CSR support is still prevailing.